web_widget_slick/static/src/js/web_widget_slick.js

```diff
@@ -0,0 +1,109 @@
+/* Copyright 2016-2017 LasLabs Inc.
+ * License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html). */
+
+odoo.define('web_widget_slick.slick_widget', function(require){
+  "use strict";
+
+  var core = require('web.core');
+  var AbstractManyField = require('web.form_relational').AbstractManyField;
+
+  var FieldSlickImages = AbstractManyField.extend({
+
+    widget_class: 'o_slick',
+    template: 'FieldSlickImages',
+    $slick: null,
+    no_rerender: true,
+    loading: [],
+    loaded: 0,
+
+    events: {
+        'mousedown img': function(e) {
+            e.preventDefault();
+        },
+        'touchstart img': function(e) {
+            e.preventDefault();
+        },
+        // Triggering a resize of the slick element on the lazyLoaded event
+        // prevents the carousel from appearing empty when page loads
+        'lazyLoaded': function() {
+            $(window).trigger('resize');
+        }
+    },
+
+    defaults: {
+      lazyLoad: 'ondemand',
+      fieldName: 'datas',
+      modelName: 'ir.attachment',
+      slidesToShow: 3,
+      slidesToScroll: 1,
+      swipeToSlide: true,
+      dots: true,
+      infinite: true,
+      speed: 500,
+      arrows: true,
+      responsive: [
+        {
+          breakpoint: 1024,
+          settings: {
+            slidesToShow: 2,
+            slidesToScroll: 1
+          }
+        },
+        {
+          breakpoint: 600,
+          settings: {
+            slidesToShow: 1,
+            slidesToScroll: 1
+          }
+        }
+        // You can unslick at a given breakpoint now by adding:
+        // settings: "unslick"
+        // instead of a settings object
+      ]
+    },
+
+    init: function(field_manager, node) {
+      this._super(field_manager, node);
+      this.options = _.defaults(this.options, this.defaults);
+    },
+
+    destroy_content: function() {
+      var self = this;
+      if (this.$slick) {
+        var $imgs = this.$el.find('img');
+        $imgs.each(function(idx, val){
+          self.$slick.slick('slickRemove', idx);
```

Comment:
Move this into a method `slickRemove`, delete the `var self = this;` above, and use proxy to send to destroy:

```
$imgs.each($.proxy(this.slickRemove, this));
```